ai-policy blog (#1248)
ropensci/roweb3: content/blog/2026-03-01-ai-policy/index.md

---
title: rOpenSci's updated policies on AI usage
author:
- Mark Padgham
date: "2026-03-01"
slug: "ropensci-ai-policy"
categories: []
tags:
- governance
- Software Peer Review
- community
- maintenance
- rOpenSci team
description: "rOpenSci's policy on the use of generative AI tools, with proposed updates to documentation and procedures for authors submitting software for review, for editors, and for reviewers."
editor:
params:
  doi: ""
---

The advent of large language models (LLMs) and generative AI tools has changed the nature of software development and of software review.

> To accommodate these changes, rOpenSci is updating its policies to clarify our expectations for authors and reviewers using generative AI tools.

Reviewer: > for authors and reviewers using generative AI tools.
Missing editors?

> Similar updates have already been implemented by both the [Journal of Open Source Software](https://blog.joss.theoj.org/2026/01/preparing-joss-for-a-generative-ai-future) and [pyOpenSci](https://www.pyopensci.org/blog/generative-ai-peer-review-policy.html), and we are indebted to the work of all those who contributed to these policy statements.

Reviewer: I would move text around a bit, bringing [the last sentence of the section](https://github.com/ropensci/roweb3/pull/1248/changes#r2836841703) here, and pushing the current sentence further down (maybe split the two clauses -- using the clause about pypensci when developing the context, and the clause about contributors to the acknowledgements section.

Concretely, the shuffling I propose results in this first paragraph, which sets a clear and positive lead for a "lead-development" narrative structure:

> The advent of large language models (LLMs) and generative AI tools has
> changed the nature of software development and of software review. To
> accommodate these changes, rOpenSci is updating its policies to clarify
> our expectations for authors and reviewers using generative AI tools. Our
> policy updates are not intended to restrict use of generative AI tools,
> rather we aim to encourage the rOpenSci community to use these tools in
> ways that encourage, rather than diminish, healthy community interactions.
> The following describes changes we will make to our requirements for
> authors, for editors, and for reviewers.

There are nevertheless many downsides to generative AI tools, as explored in the [pyOpenSci blog post](https://www.pyopensci.org/blog/generative-ai-peer-review-policy.html) which emphasises the potential for:

- Incorrect outputs
- Misleading time benefits
- Environmental impacts
- Impacts on learning
- Amplification of bias
- Infringement of copyright

There are many other downsides beyond those, such as increases in [social and economic inequality](https://pluralistic.net/2025/12/05/pop-that-bubble/#u-washington).
We hope anybody using generative AI tools to prepare software for our peer-review program takes these kinds of downsides into consideration in deciding how, when, and where to use such tools.

Generative AI tools can nevertheless contribute to producing and maintaining quality software.
However, rOpenSci’s software peer review program is designed not only to evaluate software quality, but also to act as an engine of collaboration and community building through healthy interactions between authors, reviewers, and contributors.

> Our policy updates are not intended to restrict use of generative AI tools, rather we aim to encourage the rOpenSci community to use these tools in ways that encourage, rather than diminish, healthy community interactions.

Reviewer: This sentence seems key to me. It says what the entire article does and sets a positive tone that invites reading it. I would move it to the opening paragraph to set the lead for the narrative.

## rOpenSci's new policies

Software peer-review at rOpenSci is designed to improve and promote the quality of research software, to facilitate good development practices among researchers, to foster collaboration, and to curate a community of practice.
Human interactions are a vital part of reviewing and maintaining rOpenSci software.
Software review aims to provide authors with feedback on their practices and choices, and not just on the lines of code they or an AI generated. We also value the time and effort of reviewers, and want them to be able to focus on the practices and choices of authors.

> Reviewers may use AI tools, but their reviews should still reflect their own choices and judgments.

Reviewer 2: Indeed. Of course there are cases where reviewer might be more knowledgeable than authors in some aspects, but at least if reviewer gives some feedback the authors should be able to understand it (*) and not just feed that to LLM as well... Naturally same for reviews as well.

I'm fine with _peer_-reviewing, but I feel less interested in reviewing heavily AI-generated code. Especially in an extreme case were package author(?) would externalize validating the AI-generated code they don't understand to others via peer-review... 

(*) unless it is reviewer 2.

In the words of the [JOSS blog post](https://blog.joss.theoj.org/2026/01/preparing-joss-for-a-generative-ai-future), this demands some degree of moving away from software-based "proxies for effort", to deeper consideration of the contexts in which software has been produced, and will be used.

Software reviews at rOpenSci have always been [constructive and non adversarial](https://ropensci.org/software-review/), and commonly involve extensive exchanges between authors, reviewers, and editors.
Software reviewed at rOpenSci is generally given deep and general consideration, commonly extending to the areas emphasised by JOSS's updated [scope definition](https://joss.theoj.org/about#ai-policy), and we therefore see no need to explicitly change either the scope or focus of review.
The following describes changes we will make to our requirements for authors, for editors, and for reviewers.

### For authors

It is important to us that authors who do not use generative AI tools are able to submit as always, with no extra requirements.

> Only those who use such tools will have to check a box in our submission templates, and then describe their use of generative AI tools.

Reviewer: I'm afraid these two sentences create two camps that might not be so binary. Within the "camp" of those who use AI I thing there is a wide range. 

One alternative phrasing, again grounded in values could be something like this:

We value acknowledging the external contributions that enhance our work. In the same way authors are expected to acknowledge who are the humans that helped them and how they helped, we also expect a genuine acknowledgement of what non-human forms of intelligence helped them and how.

Also we value learning and building systems. If you share your workflow, we can all learn and build custom tools that will benefit our entire community.

We will also request that authors submitting software which includes “agents.md” files or equivalent provide explicit links to those files, and describe their usage in, and influence on, the development of their software.

We note that these kinds of files effectively aim to be for generative AI tools what rOpenSci’s [required “CONTRIBUTING.md” files](https://devguide.ropensci.org/maintenance_collaboration.html#contributing-guide) have always been for human contributors.
We intend to maintain our requirement for “CONTRIBUTING.md” files, as any generative AI tools can also be directed to treat that as an “agents.md” file.

This is our intended addition to the templates for authors submitting software for review:

```
## Use of Generative AI

- [ ] Generative AI tools were used to produce some of the material in this submission.

If so, please describe: <!-- refer to our AI policies at http:// ... -->
If your repository includes an "agents.md" file or equivalent, please provide a link, and describe how this has been used in the development of your package.

> - [ ] I/we request that reviewers not use any AI tools as part of their review.

Reviewer 3: Given that:

- we already say that reviewers can use AI to navigate repos but must write their own review 
- code on GitHub will surely have already been ingested into countless AIs

what objections do we anticipate authors would have that we want to mitigate here? 

I worry that this starts to add a lot of logistical complexity if we are juggling both author/reviewer tolerances for author/reviewers having used AI. It might almost limit our ability to continue to enhance the automated repo checks since there is ultimately a somewhat fine line between algorithmic checks and heuristics that we have now and what different people might construe as "AI".

```

> ### For editors

Reviewer: I think this section is missing guidelines for what uses of AI are acceptable from editors themselves. Again, if we spell out values and derive heuristics, they should be general enough that should guide all three roles -- authors, reviewers and editors.

---
title: rOpenSci's updated policies on AI usage
author:
- Mark Padgham
date: "2026-03-01"
slug: "ropensci-ai-policy"
categories: []
tags:
- governance
- Software Peer Review
- community
- maintenance
- rOpenSci team
description: "rOpenSci's policy on the use of generative AI tools, with proposed updates to documentation and procedures for authors submitting software for review, for editors, and for reviewers."
editor:
- Mauro Lepore
params:
  doi: ""
---

The advent of large language models (LLMs) and generative AI tools has changed the nature of software development and of software review.
To accommodate these changes, rOpenSci is updating its policies to clarify our expectations for use of generative AI tools by authors, reviewers, and editors.

> Our policy updates are not intended to restrict use of generative AI tools, rather we aim to encourage the rOpenSci community to use these tools in ways that encourage, rather than diminish, healthy community interactions.

Reviewer 4: This sentence is too long which is unfortunate so early in the post.

---
title: "Software Review in the Era of AI: What We Are Testing at rOpenSci"
author:
- Mark Padgham
- Noam Ross
date: "2026-03-01"
slug: "ropensci-ai-policy"
categories: []

> tags:

Reviewer 5: Do we have an AI tag? Do we need one?

---
title: "Software Review in the era of AI: What we are testing at rOpenSci"
author:
- Mark Padgham
- Noam Ross

> date: "2026-03-01"

Reviewer 5: If you decide to publish on Thursday, make sure to update the date here and in the folder name

We can not know the eventual impact that generative AI tools may have on software review processes and practices at rOpenSci, and can ask only for an utmost degree of transparency from all involved.

We now describe specific updates to both our practices and policies for authors, for editors, and for reviewers.
Software review aims to provide authors with feedback on their practices and choices, and not just on the lines of code they or an AI generated.
We also value the time and effort of editors and reviewers, and want them to be able to focus on the practices and choices of authors.

### Practical application: for authors

It is important to us that authors who do not use generative AI tools are able to submit as always, with no extra requirements.
Those who use such tools, though, will need to support the process by sufficiently describing their use of generative AI to enable decision-making, and also affirm all generated material has been carefully reviewed by the authors.
We want these descriptions to help reviewers and editors to understand how they can best review code and provide useful feedback.

> Our submission template will now have a required section for this, and the [Guide for Authors](https://devguide.ropensci.org/softwarereview_author.html) will have additional guidance.

Reviewer 5: Will this be live when we publish the blog post? Otherwise I'd revise as follows:
```suggestion
We will be modifying our submission template to include this required section, and will be updating the [Guide for Authors](https://devguide.ropensci.org/softwarereview_author.html) with additional guidance.
```
